demos/monge_ampere_helmholtz.py

```diff
@@ -0,0 +1,208 @@
+from firedrake.pyplot import tripcolor
+
+fig, axes = plt.subplots()
+contours = tripcolor(u_h, axes=axes)
+fig.colorbar(contours)
+plt.savefig("monge_ampere_helmholtz-initial_solution.jpg")
+
+# .. figure:: monge_ampere_helmholtz-initial_solution.jpg
+#    :figwidth: 60%
+#    :align: center
+#
+# As in the Helmholtz demo, we can compute the L2-norm of the numerical error:
+
+error = u_h - u_exact(mesh)
+print("L2-norm error on initial mesh:", sqrt(assemble(dot(error, error) * dx)))
+
+# We will now try to use mesh movement to optimize the mesh to reduce
+# this numerical error. A good indicator for where resolution is required
+# is to look at the curvature of the solution which can be expressed
+# in terms of the norm of the Hessian. A monitor function
+# that targets high resolution in places of high curvature then looks like
+#
+# .. math::
+#
+#    m = 1 + \alpha \frac{H(u_h):H(u_h)}{\max_{{\bf x}\in\Omega} H(u):H(u)}
+#
+# where :math:`:` indicates the inner product, i.e. :math:`\sqrt{H:H}` is the Frobenius norm
+# of :math:`H`. We have normalized such that the minimum of the monitor function is one (where
+# the error is zero), and its maximum is :math:`1 + \alpha` (where the curvature is maximal). This
+# means we can select the ratio between the largest area and smallest area triangle in the
+# moved mesh as :math:`1+\alpha`.
+#
+# In the following implementation we use the exact solution :math:`u_{\text{exact}}` which we
+# have as a symbolic UFL expression, and thus we can also obtain the Hessian symbolically as
+# :code:`grad(grad(u_exact))`. To compute its maximum norm however we do interpolate it
+# to a P1 function space `V` and take the maximum of the array of nodal values.
+
+alpha = Constant(5.0)
+
+
+def monitor(mesh):
+    V = FunctionSpace(mesh, "CG", 1)
+    Hnorm = Function(V, name="Hnorm")
+    H = grad(grad(u_exact(mesh)))
+    Hnorm.interpolate(sqrt(inner(H, H)))
+    Hnorm_max = Hnorm.dat.data.max()
+    m = 1 + alpha * Hnorm / Hnorm_max
+    return m
+
+
+fig, axes = plt.subplots()
+m = Function(u_h, name="monitor")
+m.interpolate(monitor(mesh))
+contours = tripcolor(m, axes=axes)
+fig.colorbar(contours)
+plt.savefig("monge_ampere_helmholtz-monitor.jpg")
+fig, axes = plt.subplots()
+
+# .. figure:: monge_ampere_helmholtz-monitor.jpg
+#    :figwidth: 60%
+#    :align: center
+#
+# As in the `previous Monge-Ampère demo <./monge_ampere1.py.html>`__, we use the
+# MongeAmpereMover to perform the mesh movement based on this monitor. We need
+# to provide the monitor as a callback function that takes the mesh as its
+# input just as we have defined it above. During the iterations of the mesh
+# movement process the monitor will then be re-evaluated in the (iteratively)
+# moved mesh nodes so that, as we improve the mesh, we can also more accurately
+# express the monitor function in the desired high-resolution areas.
+
+mover = MongeAmpereMover(mesh, monitor, method="quasi_newton")
+mover.move()
```

Comment:
What do you think about including what's printed to screen in a separate code block? i.e. showing the convergence of the different metrics. We do this in Thetis demos.

Reply:
good idea. Out of interest why do we refer to the standard deviation of cell areas as "equidistribution" ?

Reply:
Hm good point, not sure how that ended up in there. Standard deviation would be a better term. We can either fix it here or open an issue for it, whichever you prefer.

Reply:
Any objection to UTF -8 in terminal output? "Standard Deviation" is a bit long, and not entirely accurate - all other terminology (https://en.wikipedia.org/wiki/Coefficient_of_variation) is a bit obscure so I thought to clarify with σ/μ

Reply:
Fine by me. Nice! Even better.